data/src/gadm/preprocess_gadm.py

```diff
@@ -7,150 +7,193 @@
+    gdf[area_col] = gdf.to_crs(epsg=6933).geometry.area / 10000
+    gdf.loc[gdf[code_col].isin(country_size_ha.keys()), area_col] = gdf[code_col].map(
+        country_size_ha
     )
-    return simplified_gdf
+    return gdf
 
 
-# 3. Join geometries with CSV data
-def join_with_csv(geom_gdf, csv_path, key="GID_0"):
-    csv_data = pd.read_csv(csv_path)
-    merged_gdf = geom_gdf.merge(csv_data, left_on=key, right_on="country_code", how="left")
-    # Drop the rows with missing country names
-    merged_gdf = merged_gdf.dropna(subset=["country"])
-    # Save the merged data
-    merged_gdf.to_file("../../raw_data/gadm/gadm_410-levels_ADM_0_merged.geojson", driver="GeoJSON")
-    print("Merged data saved at: ../../raw_data/gadm/gadm_410-levels_ADM_0_merged.geojson")
+def prepare_adm_datasets(adm0: gpd.GeoDataFrame, adm1: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    """
+    Merge the two datasets on the 'GID_0' and 'GID_1' columns. concatenating countries with Macao & Hongkong
+    """
+    adm1_prep = (
+        adm1[adm1["NAME_1"].isin(["Hong Kong", "Macau"])][["GID_1", "NAME_1", "geometry"]]
+        .rename(columns={"GID_1": "country_code", "NAME_1": "country"})
+        .pipe(lambda df: df.assign(country_code=df.country_code.str.replace("CHN.", "")))
+    )
 
-    return merged_gdf
+    return pd.concat(
+        [
+            adm0.rename(columns={"COUNTRY": "country", "GID_0": "country_code"}),
+            adm1_prep,
+        ],
+        ignore_index=True,
+    )[["geometry", "country_code"]]
 
 
 # 4. Generate SQL INSERT statement
-def generate_sql_insert(row):
+def generate_row_sql(row: pd.Series) -> str:
     sql_template = """
-    INSERT INTO public.countries
-    (code, name, continent, region_1, region_2, numeric_code, hdi, geometry)
-    VALUES ('{code}', '{name}', '{continent}'::public.countries_continent_enum, '{region_1}', ''{region_2}'', {numeric_code}, {hdi}, '{geometry}');
-    """  # noqa: E501
-
-    geometry_wkt = row["geometry"].wkt  # Get WKT representation of the geometry
-    sql = sql_template.format(
+    ('{code}', '{name}', '{continent}'::public.countries_continent_enum, '{region_1}', ''{region_2}'', {numeric_code}, {hdi}, ST_GeomFromWKB(decode({geometry}, 'hex'), 4326), {area_ha})
+    """
+    geometry_wkb = row["geometry"].wkb.hex()  # Get WKb representation of the geometry
+    sql_row = sql_template.format(
         code=row["country_code"],
         name=row["country"],
         continent=row["continent_id"],
         region_1=row["region_1"] if pd.notnull(row["region_1"]) else "NULL",
         region_2=row["region_2"] if pd.notnull(row["region_2"]) else "NULL",
         numeric_code=row["numeric"],
         hdi=row["hdi_id"] if pd.notnull(row["hdi_id"]) else "NULL",
-        geometry=geometry_wkt,
+        geometry=geometry_wkb,
+        area_ha=row["area_ha"],
     )
-    return sql
+    return sql_row
+
+
+def generate_sql_insert(rows: str) -> str:
+    return f"""
+    INSERT INTO public.countries
```

Comment:
I thought we were using UPSERT?

Is it safe to only update the area (hardcoded or precomputed) and the geom:

Reply:
As far as I know an upset in Postgres is resolve as an insert on conflict statement.